### ISSUE TITLE
Seed not working with StableDiffusionXLKDiffusionPipeline

### ISSUE THREAD

**Issue by m-pektas** [huggingface/diffusers#7421]
### Describe the bug

When I get an inference with the same seed, ı need to get the same result. It is important for reproducibility. But when I used StableDiffusionXLKDiffusionPipeline, each inference produced different results.

### Reproduction

StableDiffusionXLKDiffusionPipeline Example
```python
import torch
from diffusers import AutoPipelineForText2Image, StableDiffusionXLKDiffusionPipeline
seed = 3692506381580788093
pipeline = StableDiffusionXLKDiffusionPipeline.from_pretrained("stabilityai/stable-diffusion-xl-base-1.0", torch_dtype=torch.float16).to("cuda")
pipeline.set_scheduler("sample_dpmpp_2m_sde")
prompt = "professional photography, man, looking at camera, at Santorini,"
generator = torch.Generator(device="cuda").manual_seed(seed)
image = pipeline(generator=generator, prompt=prompt, num_inference_steps=40, use_karras_sigmas=True).images[0]
image.save("kpipe.png")
```

AutoPipelineForText2Image Example
```python
import torch
from diffusers import AutoPipelineForText2Image, StableDiffusionXLKDiffusionPipeline
seed = 3692506381580788093
pipeline = AutoPipelineForText2Image.from_pretrained("stabilityai/stable-diffusion-xl-base-1.0", torch_dtype=torch.float16).to("cuda")
prompt = "professional photography, man, looking at camera, at Santorini,"
generator = torch.Generator(device="cuda").manual_seed(seed)
image = pipeline(generator=generator, prompt=prompt, num_inference_steps=40).images[0]
image.save("default.png")
```

##  Results

![seed](https://github.com/huggingface/diffusers/assets/15729405/132f5218-6367-4481-a7e0-cbb169768e5c)

### System Info

Python 3.10.9

diffusers                 0.27.0
torch                     2.2.1

**Comment by m-pektas** [comment]
I found [this](https://github.com/huggingface/diffusers/pull/6447#issuecomment-1880621932) message in #6447 . When I add  `noise_sampler_seed=True`  as follows, the problem was solved.

```python
# before
image = pipeline(generator=gen, prompt=prompt, negative_prompt=nprompt, num_inference_steps=40, use_karras_sigmas=True).images[0]

#after
image = pipeline(generator=gen, prompt=prompt, negative_prompt=nprompt, num_inference_steps=40, use_karras_sigmas=True, noise_sampler_seed=True).images[0]

```

**Comment by standardAI** [comment]
Why `True`? `True` is only one thing why don't you give a `seed` there?

**Comment by m-pektas** [comment]
Actually, I just looked and saw that it takes seed directly. It is my fault. I didn't pay attention to the data type when trying to quickly try it :) I am fixing the above command.
